[GTK][WPE] Enable ENABLE_WEBGL2 compile-time guard when ANGLE WebGL is used

zdobersek opened the pull request:
#### 4123d040701fca819a21fa8567ea16f93101c360
<pre>
[GTK][WPE] Enable ENABLE_WEBGL2 compile-time guard when ANGLE WebGL is used
<a href="https://bugs.webkit.org/show_bug.cgi?id=243390">https://bugs.webkit.org/show_bug.cgi?id=243390</a>

Reviewed by Michael Catanzaro.

WebGL2 support gets much easier when running WebGL on top of ANGLE. Besides the
necessary runtime check there&apos;s still the ENABLE_WEBGL2 compile-time guard that
needs enabling.

For the GTK and WPE ports, the port-default ENABLE_WEBGL2 value is now ON. With
the existing dependency on the USE_ANGLE_WEBGL option, ENABLE_WEBGL2 will be
enabled when USE_ANGLE_WEBGL is turned on, and will be disabled otherwise.

* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:

Canonical link: <a href="https://commits.webkit.org/253023@main">https://commits.webkit.org/253023@main</a>
</pre>

mcatanzaro commented on Source/cmake/OptionsGTK.cmake:
```diff
@@ -112,6 +111,10 @@ else ()
 
+if (USE_ANGLE_WEBGL)
+    WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_WEBGL2 PRIVATE ON)
+endif ()
```
This is not ideal because the values of the build options like `USE_ANGLE_WEBGL` are not finalized until after `WEBKIT_OPTION_END`, which is where the depends/conflicts get applied. But after that point would be too late.

There's no great way to do this, even though it should be simple. You _could_ have `ENABLE_WEBGL2` "default" to `ON` and let it get turned off by default by the dependency on `USE_ANGLE_WEBGL`, which is not enabled by default. But that's confusing and awkward.

Alternative: perhaps `USE_ANGLE_WEBGL` will be ready for buildbots soon? If it's ready for bots but not for users, then you could make them both depend on `ENABLE_EXPERIMENTAL_FEATURES`.

Alternative: remove the build flags, since they're not needed to guard any system dependencies? That's probably a lot more than you were signing up for, though.

mcatanzaro reviewed (approved):
Confusing, since it will actually default to OFF, but this is temporary until you enable `USE_ANGLE_WEBGL`, and it's hard to think of better ways to do this.

zdobersek committed:
[GTK][WPE] Enable ENABLE_WEBGL2 compile-time guard when ANGLE WebGL is used
https://bugs.webkit.org/show_bug.cgi?id=243390

Reviewed by Michael Catanzaro.

WebGL2 support gets much easier when running WebGL on top of ANGLE. Besides the
necessary runtime check there's still the ENABLE_WEBGL2 compile-time guard that
needs enabling.

For the GTK and WPE ports, the port-default ENABLE_WEBGL2 value is now ON. With
the existing dependency on the USE_ANGLE_WEBGL option, ENABLE_WEBGL2 will be
enabled when USE_ANGLE_WEBGL is turned on, and will be disabled otherwise.

* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:

Canonical link: https://commits.webkit.org/253023@main
